File: saadmob/src/main/java/tv/superawesome/plugins/publisher/admob/SAAdMobInterstitialCustomEvent.java
package tv.superawesome.plugins.publisher.admob;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.google.android.gms.ads.AdError;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.mediation.MediationAdRequest;
import com.google.android.gms.ads.mediation.customevent.CustomEventInterstitial;
import com.google.android.gms.ads.mediation.customevent.CustomEventInterstitialListener;

import tv.superawesome.lib.sasession.defines.SAConfiguration;
import tv.superawesome.sdk.publisher.SAInterface;
import tv.superawesome.sdk.publisher.SAInterstitialAd;
import tv.superawesome.sdk.publisher.SAOrientation;

/**
 * This adapter will be retired and instead use `SAAdMobAdapter` for all ad types
 */
public class SAAdMobInterstitialCustomEvent implements CustomEventInterstitial {

    private Context context = null;
    private Integer loadedPlacementId = 0;

    @Override
    public void requestInterstitialAd(@NonNull final Context context, @NonNull final CustomEventInterstitialListener listener, String s, @NonNull MediationAdRequest mediationAdRequest, Bundle bundle) {

        // save the context
        this.context = context;

        // set values
        if (bundle != null) {
            SAInterstitialAd.setConfiguration(SAConfiguration.fromOrdinal(bundle.getInt(SAAdMobExtras.kKEY_CONFIGURATION)));
            SAInterstitialAd.setTestMode(bundle.getBoolean(SAAdMobExtras.kKEY_TEST));
            SAInterstitialAd.setParentalGate(bundle.getBoolean(SAAdMobExtras.kKEY_PARENTAL_GATE));
            SAInterstitialAd.setBumperPage(bundle.getBoolean(SAAdMobExtras.kKEY_BUMPER_PAGE));
            SAInterstitialAd.setBackButton(bundle.getBoolean(SAAdMobExtras.kKEY_BACK_BUTTON));
            SAInterstitialAd.setOrientation(SAOrientation.fromValue(bundle.getInt(SAAdMobExtras.kKEY_ORIENTATION)));
        }

        SAInterstitialAd.setListener((SAInterface) (placementId, event) -> {
            switch (event) {
                case adLoaded:
                    loadedPlacementId = placementId;
                    listener.onAdLoaded();
                    break;

                case adEmpty:
                case adFailedToLoad:
                    listener.onAdFailedToLoad(new AdError(AdRequest.ERROR_CODE_NO_FILL, "", ""));
                    break;

                case adAlreadyLoaded:
                case adEnded:
                    break;
                case adShown:
                    listener.onAdOpened();
                    break;

                case adFailedToShow:
                    listener.onAdFailedToLoad(new AdError(AdRequest.ERROR_CODE_INTERNAL_ERROR, "", ""));

                    break;

                case adClicked:
                    listener.onAdClicked();
                    listener.onAdLeftApplication();
                    break;

                case adClosed:
                    listener.onAdClosed();
                    break;

            }
        });

        try {
            int placementId = Integer.parseInt(s);
            SAInterstitialAd.load(placementId, context);
        } catch (NumberFormatException e) {
            listener.onAdFailedToLoad(new AdError(AdRequest.ERROR_CODE_INVALID_REQUEST, "", ""));
        }
    }

    @Override
    public void showInterstitial() {
        SAInterstitialAd.play(loadedPlacementId, context);
    }

    @Override
    public void onDestroy() {
        // do nothing
    }

    @Override
    public void onPause() {
        // do nothing
    }

    @Override
    public void onResume() {
        // do nothing
    }
}

File: saadmob/src/main/java/tv/superawesome/plugins/publisher/admob/SAAdMobInterstitialCustomEvent.kt
package tv.superawesome.plugins.publisher.admob

import android.content.Context
import android.os.Bundle
import com.google.android.gms.ads.AdError
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.mediation.MediationAdRequest
import com.google.android.gms.ads.mediation.customevent.CustomEventInterstitial
import com.google.android.gms.ads.mediation.customevent.CustomEventInterstitialListener
import tv.superawesome.sdk.publisher.SAEvent
import tv.superawesome.sdk.publisher.SAInterface
import tv.superawesome.sdk.publisher.SAInterstitialAd.*
import tv.superawesome.sdk.publisher.SAOrientation

@Deprecated(
    "Kept for backward compatibility reasons",
    ReplaceWith("SAAdMobAdapter")
)
class SAAdMobInterstitialCustomEvent : CustomEventInterstitial {
    private var context: Context? = null
    private var loadedPlacementId = 0
    override fun requestInterstitialAd(
        context: Context,
        listener: CustomEventInterstitialListener,
        placement: String?,
        mediationAdRequest: MediationAdRequest,
        bundle: Bundle?
    ) {

        // save the context
        this.context = context

        // set values
        if (bundle != null) {
            setParentalGate(bundle.getBoolean(SAAdMobExtras.kKEY_PARENTAL_GATE))
            setBumperPage(bundle.getBoolean(SAAdMobExtras.kKEY_BUMPER_PAGE))
            setBackButton(bundle.getBoolean(SAAdMobExtras.kKEY_BACK_BUTTON))
            setOrientation(SAOrientation.fromValue(bundle.getInt(SAAdMobExtras.kKEY_ORIENTATION)))
        }
        setListener(SAInterface { placementId: Int, event: SAEvent? ->
            when (event) {
                SAEvent.adLoaded -> {
                    loadedPlacementId = placementId
                    listener.onAdLoaded()
                }
                SAEvent.adEmpty, SAEvent.adFailedToLoad -> listener.onAdFailedToLoad(
                    AdError(
                        AdRequest.ERROR_CODE_NO_FILL,
                        "",
                        ""
                    )
                )
                SAEvent.adShown -> listener.onAdOpened()
                SAEvent.adFailedToShow -> listener.onAdFailedToLoad(
                    AdError(
                        AdRequest.ERROR_CODE_INTERNAL_ERROR,
                        "",
                        ""
                    )
                )
                SAEvent.adClicked -> {
                    listener.onAdClicked()
                    listener.onAdLeftApplication()
                }
                SAEvent.adClosed -> listener.onAdClosed()
                null -> return@SAInterface
                else -> Unit // Unused
            }
        })
        try {
            val placementId = placement!!.toInt()
            load(placementId, context)
        } catch (e: NumberFormatException) {
            listener.onAdFailedToLoad(AdError(AdRequest.ERROR_CODE_INVALID_REQUEST, "", ""))
        }
    }

    override fun showInterstitial() {
        play(loadedPlacementId, context!!)
    }

    override fun onDestroy() {
        // do nothing
    }

    override fun onPause() {
        // do nothing
    }

    override fun onResume() {
        // do nothing
    }
}
